cmd/notation/sign.go

```diff
@@ -149,3 +128,28 @@ func prepareSigningContent(ctx context.Context, opts *signOpts) (notation.Descri
 		PluginConfig: pluginConfig,
 	}, nil
 }
+
+func pushSignature(ctx context.Context, opts *SecureFlagOpts, ref string, sig []byte) (notation.Descriptor, error) {
+	// initialize
+	sigRepo, err := getSignatureRepository(opts, ref)
+	if err != nil {
+		return notation.Descriptor{}, err
+	}
+	manifestDesc, err := getManifestDescriptorFromReference(ctx, opts, ref)
+	if err != nil {
+		return notation.Descriptor{}, err
+	}
+
+	// core process
+	// pass in nonempty annotations if needed
+	sigMediaType, err := envelope.SpeculateSignatureEnvelopeFormat(sig)
```

Comment:
Can we use envelope type passed to the sign command instead of speculating it manually?

Reply:
This file is no longer in the scope of this PR. Just rebased.